packages/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts

```diff
@@ -1,147 +1,55 @@
+  const runtimePlugin: FunctionRuntimeLifecycleManager = (await context.amplify.loadRuntimePlugin(
+    context,
+    layerConfig.runtimes[0].runtimePluginId,
+  )) as FunctionRuntimeLifecycleManager;
 
-    const libGlob = glob.sync(path.join(layerDirPath, 'lib', '*'));
-    const optPath = path.join(layerDirPath, 'opt');
-
-    let conflicts: string[] = [];
-    libGlob.forEach(lib => {
-      const basename = path.basename(lib);
-      if (fs.pathExistsSync(path.join(optPath, basename))) {
-        conflicts.push(basename);
-      }
-    });
-    if (conflicts.length > 0) {
-      const libs = conflicts.map(lib => `"/${lib}"`).join(', ');
-      const plural = conflicts.length > 1 ? 'ies' : 'y';
-      context.print.warning(
-        `${libs} sub director${plural} found in both "/lib" and "/opt". These folders will be merged and the files in "/opt" will take precedence if a conflict exists.`,
-      );
-    }
-
-    zip.pipe(output);
-    [optPath, ...libGlob]
-      .filter(folder => fs.lstatSync(folder).isDirectory())
-      .forEach(folder =>
-        zip.directory(
-          folder,
-          // opt files need to be in the root of the zipped dir
-          path.basename(folder) === 'opt' ? false : path.basename(folder),
-        ),
-      );
-    zip.finalize();
-  });
-}
-
-// Check hash results for content changes, bump version if so
-async function ensureLayerVersion(context: $TSContext, layerName: string, previousHash: string) {
-  const currentHash = await hashLayerVersionContents(getLayerPath(layerName));
-  const isNewVersion = previousHash !== currentHash;
-  if (isNewVersion) {
-    if (previousHash) {
-      context.print.success(`Content changes in Lambda layer ${layerName} detected.`);
-      context.print.warning('Note: You need to run "amplify update function" to configure your functions with the latest layer version.');
-    }
-  }
-
-  const layerParameters = loadStoredLayerParameters(context, layerName);
-  await updateLayerArtifacts(context, layerParameters, { layerParams: false });
-  return currentHash;
-}
-
-// wrapper around hashElement that will return an empty string if the path does not exist
-const safeHash = async (path: string, opts?: HashElementOptions): Promise<string> => {
-  if (fs.pathExistsSync(path)) {
-    return (
-      await hashElement(path, opts).catch(() => {
-        throw new Error(`An error occurred hashing directory ${path}`);
-      })
-    ).hash;
+  const previousHash = loadPreviousLayerHash(resource.resourceName);
+  const currentHash = await ensureLayerVersion(context, resource.resourceName, previousHash);
+  // prepare package request
+  const destination = path.join(distDir, 'latest-build.zip');
+  const packageRequest = {
+    env: context.amplify.getEnvInfo().envName,
+    srcRoot: layerCodePath,
+    dstFilename: destination,
+    runtime: layerConfig.runtimes[0].value,
+    lastPackageTimeStamp: resource.lastPackageTimeStamp ? new Date(resource.lastPackageTimeStamp) : undefined,
```

Comment:
Can we fix all `TimeStamp` to `Timestamp` where it is not persisted to meta already?

Reply:
changed the variable , but left those which are persisted in Meta files